solr/core/src/java/org/apache/solr/handler/ContentStreamHandlerBase.java

```diff
@@ -49,6 +56,21 @@ public void init(NamedList<?> args) {
 
   @Override
   public void handleRequestBody(SolrQueryRequest req, SolrQueryResponse rsp) throws Exception {
+    CircuitBreakerRegistry circuitBreakerRegistry = req.getCore().getCircuitBreakerRegistry();
+    if (circuitBreakerRegistry.isEnabled(RequestType.UPDATE)) {
+      List<CircuitBreaker> trippedCircuitBreakers =
+          circuitBreakerRegistry.checkTripped(RequestType.UPDATE);
+      if (trippedCircuitBreakers != null) {
+        String errorMessage = CircuitBreakerRegistry.toErrorMessage(trippedCircuitBreakers);
+        rsp.add(STATUS, FAILURE);
+        rsp.setException(
+            new SolrException(
+                SolrException.ErrorCode.SERVICE_UNAVAILABLE,
+                "Circuit Breakers tripped " + errorMessage));
+        return;
+      }
+    }
+
```

Comment:
subjective/possibly niche use case: having a `protected boolean checkCircuitBreakers(SolrQueryRequest req, SolrQueryResponse rsp) throws Exception` method called as `if (checkCircuitBreakers(req, rsp) return;` could allow customisation e.g. overriding as no-op for one update handler but not another or to log-and-ignore the tripped circuit breakers if the `req` contained some special fields or something.